addons/project_timesheet_holidays/models/account_analytic.py

```diff
@@ -55,3 +55,9 @@ def _get_favorite_project_id_domain(self, employee_id=False):
             super()._get_favorite_project_id_domain(employee_id),
             [('holiday_id', '=', False), ('global_leave_id', '=', False)],
         ])
+
+    @api.model_create_multi
+    def create(self, vals_list):
+
+        vals_list = [val for val in vals_list if self.env['hr.leave'].browse(val.get('holiday_id')).holiday_status_id.time_type != 'other']
```

Comment:
would be better to save work for the server and check if `holiday_id` is in the `vals_list` before moving to the second condition